### PR TITLE
PP-9474 Error codes for auth api and minor updates

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Added error codes for auth api
- Updated auth endpoint and renamed `continuation_token` to `one_time_token`
- Specify mandatory fields for auth api

### CODE REVIEW

File: authorisations-api.apibp [hunk 1]
```diff
@@ -94,23 +101,29 @@ Upon completion the payment will be in one of the following [states](https://doc
...
-        + `continuation_token`: `123abc123` (string, required) - the continuation token provided in the `auth_url_post` link member of the create payment API response
-        + `card_number`: `4242424242424242` (string) - 12-19 digits
-        + `cvc`: `123` (string) - 3-4 digits
-        + `expiry_date`: `02/09` (string) - 5 character string in MM/YY format
-        + `cardholder_name`: `Joanne Smith` (string, optional)
+        + `one_time_token`: `123abc123` (string, required) - the one time token provided in the `auth_url_post` link member of the create payment API response
+        + `card_number`: `4242424242424242` (string, required) - 12-19 digits
+        + `cvc`: `123` (string, required) - 3-4 digits
+        + `expiry_date`: `02/09` (string, required) - 5 character string in MM/YY format
+        + `cardholder_name`: `Joanne Smith` (string, required)
```
Comment (stephencdaly) [inline authorisations-api.apibp]: Should we explicitly state the limit of 255 characters for the cardholder_name?

File: authorisations-api.apibp [hunk 2]
```diff
@@ -94,23 +101,29 @@ Upon completion the payment will be in one of the following [states](https://doc
...
         
-+ Response 201 (application/json)
++ Response 204 (application/json)
 
-        Created
++ Response 400 (application/json)
```
Comment (stephencdaly) [inline authorisations-api.apibp:116]: Would it be good to add the 422 error code, or do we think that is implied by the rest of the API docs?

File: authorisations-api.apibp [hunk 3]
```diff
@@ -202,14 +215,27 @@ self,events and next links of a Payment
 ### Properties
 + `type`: `multipart/form-data` (string, optional) 
 + `params`: (object)
-+ `href`: `https://auth.payments.service.gov.uk/auth/<secure_token>` (string, optional)
++ `href`: `https://publicapi.payments.service.gov.uk/v1/auth/<secure_token>` (string, optional)
```
Comment (stephencdaly) [inline authorisations-api.apibp]: The `href` won't have `<secure_token>` on the end

File: authorisations-api.apibp (hunk 1 above)
Comment (kbottla) [reply authorisations-api.apibp]: yeah updated!

File: authorisations-api.apibp (hunk 2 above)
Comment (kbottla) [reply authorisations-api.apibp:116]: yeah we should add it. Updated!

**Review by stephencdaly — APPROVED** [review]
Looks good!

**Pull request merged**